### ISSUE TITLE
Debugger Is Broken When Build Directory Is No Longer Present

### ISSUE THREAD

**Issue by Shinmera** [drmeister/clasp#151]
On 7fde5b78e98bf1af8e74f3884815a81db9eeec7e.
How to reproduce: copy `build/clasp` to a machine where the build root directory does not exist, or copy it out and move the build root directory someplace else.
```
╭─linus@linuy ~/Projects/clasp/build  ‹dev*› 
╰─➤  cd ~/foo/clasp 
╭─linus@linuy ~/foo/clasp  
╰─➤  ls
bin  Contents
╭─linus@linuy ~/foo/clasp  
╰─➤  cd Contents/execs/boehm/release/bin 
╭─linus@linuy ~/foo/clasp/Contents/execs/boehm/release/bin  
╰─➤  ls
clasp  intrinsics_bitcode.sbc
╭─linus@linuy ~/foo/clasp/Contents/execs/boehm/release/bin  
╰─➤  ./clasp 
Starting Boehm cclasp 0.3-test-10-463-g7fde5b7 ... loading image... it takes a few seconds
Top level.
> (logical-pathname-translations "SYS")

((#P"SYS:**;*.*" #P"/home/linus/foo/clasp/Contents/Resources/lisp/**/*.*"))
> (foo)
;;; Warning: Condition: Undefined function named COMMON-LISP-USER::FOO

Condition of type: SIMPLE-PROGRAM-ERROR
The symbol FOO has no function bound to it

Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[117] TOP-LEVEL.
Debugger received error of type: SIMPLE-PROGRAM-ERROR
The only argument for pathname is nil
Error flushed.
Broken at frame[117] TOP-LEVEL.
Debugger received error of type: SIMPLE-PROGRAM-ERROR
The only argument for pathname is nil
Error flushed.
Broken at frame[117] TOP-LEVEL.
Debugger received error of type: SIMPLE-PROGRAM-ERROR
The only argument for pathname is nil
Error flushed.
Broken at frame[117] TOP-LEVEL.
Debugger received error of type: SIMPLE-PROGRAM-ERROR
The only argument for pathname is nil
Error flushed.
Broken at frame[117] TOP-LEVEL.
Debugger received error of type: SIMPLE-PROGRAM-ERROR
The only argument for pathname is nil
Error flushed.
Broken at frame[117] TOP-LEVEL.
Debugger received error of type: SIMPLE-PROGRAM-ERROR
The only argument for pathname is nil
Error flushed.
Broken at frame[117] TOP-LEVEL.
Debugger received error of type: SIMPLE-PROGRAM-ERROR
The only argument for pathname is nil
Error flushed.
Broken at frame[117] TOP-LEVEL.
Debugger received error of type: SIMPLE-PROGRAM-ERROR
The only argument for pathname is nil
Error flushed.
..... and so forth .....
```
It starts printing this wall of errors immediately and seemingly forever. I had to press C-c a bunch of times to get it to stop. As you can see, the logical pathname translations seem to have been figured out correctly and `Resources/lisp` does indeed contain the files too:
```
╭─linus@linuy ~/foo/clasp/Contents/Resources/lisp  
╰─➤  ls
build                   gc-builder_new.lsp      prof.lsp               swank-loader.lsp
clasp-analyzer          gc-builder-new.lsp      refactoring-tools.lsp  system.lsp
compile-commands.lsp    gc-kind.lsp             refactor.lsp           testmod.lsp
easy-modules.lsp        kernel                  serialize              tests
extract-gc-builder.lsp  local-asdf-config.lisp  serialize.lsp
gc-builder.lsp          modules                 slime
```
It looks like some where there's still stray references to the build directory that then cause this infinite chain of errors. Sadly this bug is a showstopper as it makes binary releases impossible, so we cannot release before it is fixed.

**Comment by drmeister** [comment]
I think this should be fixed in the current version.   I fixed this over the weekend.

**Comment by Shinmera** [comment]
Fix confirmed on 712260b4038b40e04d8928b7183e2af5da416a0d
